swaylock/main.c

```diff
@@ -101,12 +102,17 @@ void notify_key(enum wl_keyboard_key_state state, xkb_keysym_t sym, uint32_t cod
 			if (verify_password()) {
 				exit(0);
 			}
-			password = malloc(1024); // TODO: Let this grow
+			password_size = 1024;
+			password = malloc(password_size);
```

Comment:
The previous password should be freed before this, so it doesn't leak. (In case you make a new PR @gkbrk)